SharkCage/CageServiceInstaller/SharkCageInstallerHelper.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Microsoft.Win32;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections;
+using System.ComponentModel;
+using System.Configuration.Install;
+using System.IO;
+using System.Security.AccessControl;
+using System.Security.Cryptography;
+using System.Security.Principal;
+
+namespace SharkCageInstallerHelper
+{
+    [RunInstaller(true)]
+    public partial class SharkCageInstallerHelper : Installer
+    {
+        const string service_name = "shark-cage-service";
+
+        public override void Install(System.Collections.IDictionary state_saver)
+        {
+            base.Install(state_saver);
+        }
+
+        public override void Commit(System.Collections.IDictionary saved_state)
+        {
+            base.Commit(saved_state);
+
+            string target_dir = Context.Parameters["DP_TargetDir"].ToString();
+
+            AssimilateConfigToCurrentSystem(target_dir);
+
+            InstallAndStartService(target_dir);
+        }
+
+        public override void Rollback(IDictionary saved_state)
+        {
+            base.Rollback(saved_state);
+            UninstallService();
+        }
+
+        public override void Uninstall(System.Collections.IDictionary saved_state)
+        {
+            base.Uninstall(saved_state);
+            UninstallService();
+        }
+
+        private void AssimilateConfigToCurrentSystem(string dir_path)
+        {
+            var config_path = Path.Combine(
+                Environment.GetFolderPath(Environment.SpecialFolder.CommonDocuments),
+                "SharkCage\\CageConfigurator.sconfig");
```

Comment:
You can use `@"one\slash\only"` if you don't want to use characters which are needed to be escaped (with `@` everything is interpreted literally